frontend/src/libs/components/modal/steps/delete-account/delete-account-confirmation.tsx

```diff
@@ -0,0 +1,57 @@
+import { type FC } from 'react';
+
+import { Button } from '#libs/components/components.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+} from '#libs/hooks/hooks.js';
+import { type UserDeleteRequestDto } from '#packages/users/users.js';
+import { actions as authActions } from '#slices/auth/auth.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  onClose?: () => void;
+};
+
+const DeleteAccountConfirmation: FC<Properties> = ({ onClose }) => {
+  const { authenticatedUser } = useAppSelector(({ auth }) => {
+    return {
+      authenticatedUser: auth.authenticatedUser,
+    };
+  });
+
+  const authenticatedUserId = authenticatedUser?.id;
+
+  const dispatch = useAppDispatch();
+  const handleDelete = useCallback(
+    (payload: UserDeleteRequestDto): void => {
+      void dispatch(authActions.deleteUser(payload));
+      onClose?.();
```

Comment:
I suppose we should also handle logout for auth user after deleting

Reply:
```typescript
const deleteUser = createAsyncThunk<
  unknown,
  UserDeleteRequestDto,
  AsyncThunkConfig
>(`${sliceName}/delete-user`, async (deletePayload, { extra, dispatch }) => {
  const { authApi, storage } = extra;
  const { id } = await authApi.deleteUser(deletePayload);

  await storage.drop(StorageKey.TOKEN);

  dispatch(appActions.navigate(AppRoute.SIGN_IN));

  return id;
});
```

I did it here frontend\src\slices\auth\actions.ts

I removed the token from the user and redirected it to the sign-in page so the user can no longer sign in to the page once deleted, let me know if it is ok or if you need me to do this elsewhere.

Reply:
@v0ldemar01